### DISCUSSION TITLE
how to build a staggered structure with alternating orientations.

### DISCUSSION THREAD

**Discussion by richhongr** [marrink-lab/polyply_1.0#372] [Q&A]
Hello! Thank you for creating such a wonderful toolkit. I have already created the .ff file, and now I want to create .gro files for both a monomer and a staggered structure with alternating orientations. Could you please guide me on how to do this? Here are the commands I used to create the .gro file so far:

1. ```polyply gen_params -lib gromos53A6 -o P3HT.itp -name P3HT -seq Hter:1 P3HT:10 Hter:1```
2. ```polyply gen_coords -p P3HT.top -o melt.gro -name melt -dens 10 -back_fudge 1 -mf 0.5```

**Comment by fgrunewald** [comment]
@richhongr I'm not sure what you mean by alternating orientations; you mean head to tail vs tail to tail of the P3HT? 
@ricalessandri perhaps it the better person to help with this

**Reply by richhongr** [reply]
Thank you for your reply. Alternating orientations mean that the dihedral angle of SCCS between monomers is 180 degrees. As shown in the figure
![bithiophene](https://github.com/marrink-lab/polyply_1.0/assets/134678317/550d4eab-d5bd-4dd6-8042-e5e2f237856e)

**Reply by fgrunewald** [reply]
good question; we have been working on something like this for RNA/DNA but it's not fully implemented yet. Instead you could add an extra dihedral angle that biases this conformation. If you are not building a dense melt it should work. Perhaps as a perquisite use the following build file to make 'straighter' chains that give a better approximation to the persistence length:

```
[ bending ]
P3HT P3HT P3HT 8
``` 

Use the above as : `polyply -b build_file.bld ...` The higher you put the number the straighter the chain becomes.  

Before generating the structure you need to to modify the existing .ff files by adding the following dihedral angle:

```
[link]
resname "P3HT"
[ atoms ]
ST1 { }
CT2 { }
+CT5 { }
+ST1 { }
[ dihedrals ]
  ST1   CT2  +CT5  +ST1   1        0.0       1.80          1  {"group": "eq dihedral", "version":"5", "ifdef": "eq_polyply"} 
[ edges ]
ST1 CT2
CT2 +CT5
+CT5 +ST1
```

Then generate the itp file again, the structure and run the energy minimization but put 'defines = -Deq_polyply' on the top if the MDP file. Potentially you need to run some short MD with 'defines = -Deq_polyply' to fully get what you want.

**Reply by richhongr** [reply]
Thank you for your response! However, when I use this command:
```polyply gen_coords -p P3HT.top -o melt.gro -name melt -dens 10 -back_fudge 1 -mf 0.5 -b build_file.bld```
It throws an error:
```OSError: Can't parse line 2 in section '['bending']' because the section is unknown```
How should I resolve this issue?

**Reply by fgrunewald** [reply]
@richhongr the bending is a relatively new feature. you probably have to reinstall polyply

**Reply by richhongr** [reply]
Okay, thank you for your assistance.

**Reply by fgrunewald** [reply]
@richhongr let me know if the strategy works out; it might be useful for others too
